From 87ee60db0e780ec6c67c50137b7b7a763d610b47 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Mon, 27 Jan 2025 14:52:35 +0100
Subject: [PATCH 790/954] feat(build): add encrypted payload as FIP dependency

When using TOOL_ADD_PAYLOAD with an encrypted file as 5th parameter,
this file should be added as a dependency to FIP file creation, adding
it to FIP_DEPS.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git e838bff352af130e4029bbaa2da3aa21986ac54d feat(build): add encrypted payload as FIP dependency]
Change-Id: I69e81a333578fd894117afceb92ab0ed950932d7
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/431967
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 make_helpers/build_macros.mk | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/make_helpers/build_macros.mk b/make_helpers/build_macros.mk
index e3194a8891..507692afe3 100644
--- a/make_helpers/build_macros.mk
+++ b/make_helpers/build_macros.mk
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2015-2023, Arm Limited and Contributors. All rights reserved.
+# Copyright (c) 2015-2025, Arm Limited and Contributors. All rights reserved.
 #
 # SPDX-License-Identifier: BSD-3-Clause
 #
@@ -195,6 +195,7 @@ define TOOL_ADD_PAYLOAD
 ifneq ($(5),)
     $(4)FIP_ARGS += $(2) $(5)
     $(if $(3),$(4)CRT_DEPS += $(1))
+    $(if $(3),$(4)FIP_DEPS += $(5))
 else
     $(4)FIP_ARGS += $(2) $(1)
     $(if $(3),$(4)CRT_DEPS += $(3))
-- 
2.53.0

